src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java

```diff
@@ -153,223 +167,229 @@ public BuildStepMonitor getRequiredMonitorService() {
      */
     @Override
     public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) {
-	StorageAccountInfo strAcc = null;
-	try {
-	    final EnvVars envVars = run.getEnvironment(listener);
-
-	    // Get storage account
-	    strAcc = getDescriptor().getStorageAccount(storageAccName);
-
-	    // Resolve include patterns
-	    String expIncludePattern = Util.replaceMacro(includeFilesPattern, envVars);
-
-	    // Resolve exclude patterns
-	    String expExcludePattern = Util.replaceMacro(excludeFilesPattern, envVars);
-
-	    String expProjectName = Util.replaceMacro(projectName, envVars);
-
-	    String expContainerName = Util.replaceMacro(containerName, envVars);
-
-	    // If the include is empty, make **/*
-	    if (Utils.isNullOrEmpty(expIncludePattern)) {
-		expIncludePattern = "**/*";
-	    }
-
-	    // Exclude archive.zip by default.
-	    if (!includeArchiveZips) {
-		if (expExcludePattern != null) {
-		    expExcludePattern += ",archive.zip";
-		} else {
-		    expExcludePattern = "archive.zip";
-		}
-	    }
-
-	    // Validate input data
-	    validateData(run, listener, strAcc);
-
-	    int filesDownloaded = 0;
-	    if (downloadType == null || downloadType.equals("container")) { /*the null check is backward compatibility*/
-		    filesDownloaded += WAStorageClient.download(run, launcher, listener,
-			    strAcc, expIncludePattern, expExcludePattern, Util.replaceMacro(downloadDirLoc, envVars),
-			    flattenDirectories, workspace, expContainerName);
-	    } else {
-		    Job<?, ?> job = Jenkins.getInstance().getItemByFullName(expProjectName, Job.class);
-		    if (job != null) {
-			// Resolve download location
-			BuildFilter filter = new BuildFilter();
-			Run source = getBuildSelector().getBuild(job, envVars, filter, run);
-
-			if (source instanceof MatrixBuild) {
-				for (Run r : ((MatrixBuild) source).getExactRuns()) {
-				    filesDownloaded += downloadArtifacts(r, run, launcher, expIncludePattern, expExcludePattern, listener, strAcc, workspace);
-				}
-			} else {
-				filesDownloaded += downloadArtifacts(source, run, launcher, expIncludePattern, expExcludePattern, listener, strAcc, workspace);
-			}
-		} else {
-		    listener.getLogger().println(Messages.AzureStorageBuilder_job_invalid(expProjectName));
-		    run.setResult(Result.UNSTABLE);
-		}
-	    }
-
-	    if (filesDownloaded == 0) { // Mark build unstable if no files are
-		    // downloaded
-		    listener.getLogger().println(Messages.AzureStorageBuilder_nofiles_downloaded());
-		    run.setResult(Result.UNSTABLE);
-	    } else {
-		    listener.getLogger().println(Messages.AzureStorageBuilder_files_downloaded_count(filesDownloaded));
-	    }
-
-	} catch (Exception e) {
-	    e.printStackTrace(listener.error(Messages.AzureStorageBuilder_download_err(strAcc.getStorageAccName())));
-	    run.setResult(Result.UNSTABLE);
-	}
+        StorageAccountInfo strAcc = null;
```

Comment:
sure .. sorry for the pain .. I will make sure ... in the mean time please let me know if you find it difficult ... only major changes were in storagepublisher, storagebuilder and utils. publisher and builder are affected inside perform and databound constructors and inside dofill methods